lib/components/normal-components/Crystal_4Pin.ts

```diff
@@ -0,0 +1,52 @@
+import { crystalProps } from "@tscircuit/props"
+import {
+  type BaseSymbolName,
+  type Ftype,
+  type PolarizedPassivePorts,
+} from "lib/utils/constants"
+import { NormalComponent } from "../base-components/NormalComponent"
+import type { SourceSimpleCrystal } from "circuit-json"
+import { formatSiUnit } from "format-si-unit"
+
+export class Crystal4Pin extends NormalComponent<
+  typeof crystalProps,
+  PolarizedPassivePorts
+> {
+  // @ts-ignore
+  get config() {
+    return {
+      schematicSymbolName:
+        this.props.symbolName ?? ("crystal_4pin" as BaseSymbolName),
+      componentName: "Crystal4Pin",
+      zodProps: crystalProps,
+      sourceFtype: "simple_crystal" as Ftype,
+    }
+  }
+
+  initPorts() {
+    super.initPorts({
+      additionalAliases: {
+        pin1: ["pos", "top_left"],
+        pin2: ["neg", "top_right"],
+        pin3: ["cap1", "bottom_left"],
+        pin4: ["cap2", "bottom_right"],
```

Comment:
The alias for this probably should be `top`, `bottom`, `left` and `right`, according to the symbol image